### PR TITLE
[Web] Adds support for Flask-like "blueprints"

### PR DESCRIPTION
**Pull request by ask**
Blueprint is basically just a description of a reusable app
that you can add to your web application.

Blueprints are commonly used in most Flask-like web frameworks,
but Flask blueprints are not compatible with e.g. Sanic blueprints.

The Faust blueprint is not directly compatible with any of them,
but that should be fine.

To define a blueprint:

```py

    from faust import web

    blueprint = web.Blueprint('user')

    @blueprint.route('/', name='list')
    class UserListView(web.View):

        async def get(self, request: web.Request) -> web.Response:
            return self.json({'hello': 'world'})

    @blueprint.route('/{username}/', name='detail')
    class UserDetailView(web.View):

        async def get(self, request: web.Request) -> web.Response:
            name = request.match_info['username']
            return self.json({'hello': name})

        async def post(self, request: web.Request) -> web.Response:
            ...

        async def delete(self, request: web.Request) -> web.Response:
            ...

```

Then to add the blueprint to a Faust app you register it:

```py
    blueprint.register(app, url_prefix='/users/')
```

NOTE:

You can also create views from functions (in this case it will only
support GET):

```py
@blueprint.route('/', name='index')
async def hello(self, request):
    return self.text('Hello world')
```

.

### CODE REVIEW

**Comment by ask** [comment]
One suggestion is, maybe we should pass match_info as keyword arguments to the view,
so that 
```py
        async def get(self, request: web.Request) -> web.Response:
            name = request.match_info['username']
            return self.json({'hello': name})
```

becomes:

```
async def get(self, request: web.Request, username: str) -> web.Response:
    return self.json({'hello': username})
```

**Comment by ask** [comment]
I added the backward incompatible change that match_info is passed as additional arguments to the methods (see previous comment)

**Comment by vineetgoel** [comment]
I really like the idea of sending match_info as a keyword argument into the function.

File: faust/web/base.py [hunk 1]
```diff
@@ -34,8 +39,35 @@ def __init__(self,
...
+
+        If the provided view name has associated URL parameters,
+        those need to be passed in as kwargs, or a :exc:`TypeError`
+        will be raised.
+        """
+        try:
+            path = self.reverse_names[view_name]
+        except KeyError:
+            raise KeyError(f'No view with name {view_name!r} found')
+        return path.format(**{
```
Comment (vineetgoel) [inline faust/web/base.py]: Should we add this in an `else` block?

File: faust/web/blueprints.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        self.url_prefix = url_prefix
+
+        self.routes = []
+        self.static_routes = []
+
+    def clone(self, url_prefix: Optional[str] = None) -> BlueprintT:
+        if url_prefix is None:
+            url_prefix = self.url_prefix
+        bp = type(self)(name=self.name, url_prefix=url_prefix)
+        bp.routes = self.routes  # XXX Do not modify!!!
```
Comment (vineetgoel) [inline faust/web/blueprints.py]: Maybe make the comment more explicit. This is confusing as I though `bp.routes` should never be modified but adding more routes to the blueprint will update this.

**Pull request merged**